analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -43,7 +43,8 @@ public override ProgramState ConditionEvaluated(SymbolicContext context)
     {
         var operation = context.Operation.Instance;
         if (operation.Kind is not OperationKindEx.Literal
-            && operation.Syntax.Ancestors().Any(IsUsing) is false)
+            && operation.Syntax.Ancestors().Any(IsUsing) is false
+            && !IsConstField(operation))
```

Comment:
I wonder if we should just use `!operation.ConstantValue.HasValue` instead. Let's keep that option in mind and see if we get other annoying FPs with the field approach.